File: app/src/main/java/org/eyeseetea/malariacare/domain/boundary/repositories/IServerRepository.java
package org.eyeseetea.malariacare.domain.boundary.repositories;

import org.eyeseetea.malariacare.domain.common.ReadPolicy;
import org.eyeseetea.malariacare.domain.entity.Server;

import java.util.List;

public interface IServerRepository {
    List<Server> getAll(ReadPolicy readPolicy);
    Server getLoggedServer() throws Exception;
}

File: app/src/main/java/org/eyeseetea/malariacare/domain/boundary/repositories/IServerRepository.kt
package org.eyeseetea.malariacare.domain.boundary.repositories

import org.eyeseetea.malariacare.domain.common.Either
import org.eyeseetea.malariacare.domain.common.ReadPolicy
import org.eyeseetea.malariacare.domain.entity.Server
import org.eyeseetea.malariacare.domain.usecase.GetServerFailure

interface IServerRepository {
    fun getAll(readPolicy: ReadPolicy): List<Server>

    fun getLoggedServer(): Either<GetServerFailure, Server>

    fun save(server: Server)
}